Kernel/FileSystem/FATFS/Inode.cpp

```diff
@@ -710,18 +774,26 @@ ErrorOr<void> FATInode::add_child(Inode& inode, StringView name, mode_t mode)
     dbgln_if(FAT_DEBUG, "FATInode[{}]::add_child(): appending inode {} as \"{}\"", identifier(), inode.identifier(), name);
 
     auto entry = bit_cast<FATInode*>(&inode)->m_entry;
-    auto sfn = TRY(create_sfn_from_lfn(name));
-    auto existing_sfns = TRY(collect_sfns());
-    TRY(create_unique_sfn_for(entry, move(sfn), move(existing_sfns)));
+
+    bool valid_sfn = is_valid_sfn(name);
+
+    if (valid_sfn) {
+        TRY(encode_known_good_sfn_for(entry, name));
```

Comment:
entry isn't zero-initialized on this codepath, and encode_known_good_sfn doesn't zero out the buffers before copying over. Is that fine, or can it be a problem?

Reply:
Thanks, not sure how I missed that given I set out to specifically look for it 🤦